### PR TITLE
Store pandas dataframe as single object in data frame 

### PR DESCRIPTION
**Pull request by jan-janssen**
Converting to dictionary is terribly slow: 
```
from pyiron_base import DataContainer
from pyiron_base import Project
import pandas
with open("test.csv", "w") as f:
    f.write(",a,b\n")
    for i in range(100000): 
        f.write(str(i) + ",1,2\n")
df = pandas.read_csv("test.csv", index_col=0)
pr = Project(".")
hdf = pr.create_hdf(".", "test")
dc = DataContainer(df.to_dict())
dc.to_hdf(hdf)
```
This takes forever, while with a small change it is in the order of ms: 
```
dc = DataContainer(df)
```

### CODE REVIEW

**Comment by jan-janssen** [comment]
This fixes my issues with https://github.com/pyiron/pyiron_base/pull/463/

**Comment by pmrv** [comment]
Have you checked that this actually stores the data?  In my testing conversion from pandas is broken.

```python
df = DataFrame({'a': range(4), 'b': range(4,8), 'c': range(8,12)})
dc = DataContainer(df)
print(dc)
> DataContainer([])
dc = DataContainer(df.to_dict())
print(dc)
> DataContainer({'a': DataContainer([0, 1, 2, 3]), 'b': DataContainer([4, 5, 6, 7]), 'c': DataContainer([8, 9, 10, 11])})
```

The reason I didn't save it via pandas is that I expect people to put data into a pyiron_table that is HDF-serializable to us, but not to pandas (e.g. structures).  I'll have a look today at #463 and #559.

File: pyiron_base/table/datamining.py [hunk 1]
```diff
@@ -692,8 +690,7 @@ def from_hdf(self, hdf=None, group_name=None):
         if hdf_version=="0.3.0":
             with self.project_hdf5.open("output") as hdf5_output:
                 if "table" in hdf5_output.list_groups():
-                    data = hdf5_output["table"].to_object().to_builtin()
-                    self._pyiron_table._df = pandas.DataFrame(data)
+                    self._pyiron_table._df = pandas.read_hdf(hdf5_output.file_name, hdf5_output.h5_path + "/table")
```
Comment (pmrv) [inline pyiron_base/table/datamining.py:693]: This will break without `pytables` or not?

**Comment by jan-janssen** [comment]
> The reason I didn't save it via pandas is that I expect people to put data into a pyiron_table that is HDF-serializable to us, but not to pandas (e.g. structures). I'll have a look today at #463 and #559.

That is definitely a limitation of the current solution, currently I sure the structures as JSON objects before inserting them into the pyiron table, but that requires extra work on the user side. The advantage is the size difference. Already if I only load the first 10000 entries of the file in the example above the old code resulted in a 124MB HDF5 file. So the conversion to dictionary and storing in the Datacontainer is definitely not scalable for large pyiron tables.

File: pyiron_base/table/datamining.py (hunk 1 above)
Comment (jan-janssen) [reply pyiron_base/table/datamining.py:693]: Yes, I rely on pytables - but as long as that is the recommended solution by pandas I guess that is the only option. Alternatively we could also pickle the object and store the pickled string, but I do not think that is a better alternative.

Comment (pmrv) [reply pyiron_base/table/datamining.py:693]: Then there's no need to put the `DataFrame` into the data container at all and we can just `DataFrame.to_hdf`/`read_hdf`, I suppose.  Then we also need to put `tables` back into our `environment.yml`.

Comment (jan-janssen) [reply pyiron_base/table/datamining.py:693]: That I exactly what I do in the latest commit.

**Comment by pmrv** [comment]
I'm looking into the file size right now and it seems like an issue with how we write to HDF5 in general.  When I do a `h5repack` on the file in the example, I directly get a factor ~20 decrease in file size.

File: pyiron_base/table/datamining.py (hunk 1 above)
Comment (pmrv) [reply pyiron_base/table/datamining.py:693]: Ah, I missed that.  Looks good to me then, we can always come back to this if someone really wants to store "complex" data in their pyiron tables.

EDIT: The dependency still should be added, though.

**Review by pmrv — APPROVED** [review]

**Comment by jan-janssen** [comment]
The dependency is still included https://github.com/pyiron/pyiron_base/blob/pyiron_table_store_using_pandas/setup.py#L45

**Pull request merged**